src/blazar_tempest_plugin/tests/scenario/test_images.py

```diff
@@ -152,20 +152,25 @@ def resource_setup(cls):
                     reservation_id = inst._get_host_reservation(lease)
                     flavor = CONF.reservation.reservable_flavor_ref
                     scheduler_hints = {"reservation": reservation_id}
-                else:
-                    flavor = CONF.compute.flavor_ref
+                elif CONF.reservation.reservation_type == "kvm":
+                    flavor_id = inst._get_flavor_id(CONF.reservation.reservable_flavor_ref)
+                    lease = inst._reserve_flavor_host(flavor_id=flavor_id)
```

Comment:
nit / tempest style issue:

whenever we create any resource, it should be done via a helper method which also registers a cleanup task for that resource. Using try/except for this is an antipattern in tempest-land.

That said, it was already present here in prior commits, so not like it was introduced in this PR